Test failing on windows (see appveyor) test_unicode_project_folder

```
_________________________ test_unicode_project_folder _________________________
tests\test_unicode.py:21: in test_unicode_project_folder
    prog, _ = builder.build(pad.root)
lektor\builder.py:1098: in build
    prog.build()
lektor\build_programs.py:101: in build
    _build(artifact, self.build_artifact)
lektor\build_programs.py:91: in _build
    ctx = gen.build_artifact(artifact, build_func)
lektor\builder.py:1058: in build_artifact
    build_func(artifact)
c:\python27\lib\contextlib.py:24: in __exit__
    self.gen.next()
lektor\builder.py:803: in update
    self.finish_update(ctx)
lektor\builder.py:863: in finish_update
    ctx.referenced_virtual_dependencies.values())
lektor\builder.py:738: in _memorize_dependencies
    self._auto_deferred_update_operation(operation)
lektor\builder.py:783: in _auto_deferred_update_operation
    f(con)
lektor\builder.py:673: in operation
    source = self.build_state.to_source_filename(source)
lektor\builder.py:146: in to_source_filename
    return self.path_cache.to_source_filename(filename)
lektor\builder.py:912: in to_source_filename
    (filename, self.env.root_path))
E   ValueError: The given value (u'C:\\projects\\lektor\\tests\\\xc3\xbcnic\xc3\xb6de-project\\templates\\page.html') is not below the source folder (u'C:\\projects\\lektor\\tests\\\xfcnic\xf6de-project')
```

--- a/lektor/environment.py
+++ b/lektor/environment.py
@@ -13,14 +13,14 @@ from inifile import IniFile
 from werkzeug.urls import url_parse
 from werkzeug.utils import cached_property
 
-from lektor._compat import iteritems, string_types
+from lektor._compat import iteritems, string_types, PY2
 from lektor.context import (config_proxy, get_asset_url, get_ctx, get_locale,
     site_proxy, url_to)
 from lektor.i18n import get_i18n_block
 from lektor.markdown import Markdown
 from lektor.pluginsystem import PluginController
 from lektor.utils import (bool_from_string, format_lat_long, secure_url,
-    tojson_filter)
+    tojson_filter, is_windows)
 
 
 # Special value that identifies a target to the primary alt
@@ -215,7 +215,13 @@ class CustomJinjaEnvironment(jinja2.Environment):
         try:
             rv = jinja2.Environment._load_template(self, name, globals)
             if ctx is not None:
-                ctx.record_dependency(rv.filename)
+                filename = rv.filename
+                if PY2 and is_windows:
+                    try:
+                        filename = filename.decode('utf-8')
+                    except UnicodeDecodeError:
+                        pass
+                ctx.record_dependency(filename)
             return rv
         except jinja2.TemplateSyntaxError as e:
             if ctx is not None:

